lib/AST/PluginRegistry.cpp

```diff
@@ -111,6 +111,9 @@ PluginRegistry::loadExecutablePlugin(StringRef path) {
       childInfo->Pid, stat.getLastModificationTime(),
       childInfo->ReadFileDescriptor, childInfo->WriteFileDescriptor));
 
+  const char *dumpMessaging = ::getenv("SWIFT_DUMP_PLUGIN_MESSAGING");
+  plugin->setDumpMessaging(dumpMessaging != nullptr);
```

Comment:
> Not a compiler option because a plugin can be reused between multiple compiler invocations.

That's only true for sourcekit right? Could we have this as a property on the `PluginRegistry`? Regular compilations could use either a compilation flag or the env variable and sourcekit just the env variable?

I don't think this *really* matters though, so if you'd like to keep it the way it is that's fine too.

Reply:
Made a `PluginRegistry`'s property because accessing `getenv` for each plugin loading is necessary.
As for making this a compilation flag, I don't know, just let me keep it a env var.